pkg/storageincentives/agent.go

```diff
@@ -331,31 +329,40 @@ func (a *Agent) play(ctx context.Context) (uint8, []byte, error) {
 	}
 
 	t := time.Now()
-	a.metrics.BackendCalls.Inc()
-	block, err := a.backend.BlockNumber(ctx)
+
+	timeLimiter, err := a.getPreviousRoundTime(ctx)
 	if err != nil {
-		a.metrics.BackendErrors.Inc()
 		return 0, nil, err
 	}
 
-	previousRoundNumber := (block / a.blocksPerRound) - 1
+	sample, err := a.sampler.ReserveSample(ctx, salt, storageRadius, uint64(timeLimiter))
+	if err != nil {
+		return 0, nil, err
+	}
+	a.metrics.SampleDuration.Set(time.Since(t).Seconds())
+
+	return storageRadius, sample.Hash.Bytes(), nil
+}
+
+func (a *Agent) getPreviousRoundTime(ctx context.Context) (time.Duration, error) {
 
 	a.metrics.BackendCalls.Inc()
-	timeLimiterBlock, err := a.backend.HeaderByNumber(ctx, new(big.Int).SetUint64(previousRoundNumber*a.blocksPerRound))
+	block, err := a.backend.BlockNumber(ctx)
 	if err != nil {
 		a.metrics.BackendErrors.Inc()
-		return 0, nil, err
+		return 0, err
 	}
 
-	timeLimiter := time.Duration(timeLimiterBlock.Time) * time.Second / time.Nanosecond
+	previousRoundBlockNumber := ((block / a.blocksPerRound) - 1) * a.blocksPerRound
 
-	sample, err := a.sampler.ReserveSample(ctx, salt, storageRadius, uint64(timeLimiter))
+	a.metrics.BackendCalls.Inc()
+	timeLimiterBlock, err := a.backend.HeaderByNumber(ctx, new(big.Int).SetUint64(previousRoundBlockNumber))
 	if err != nil {
-		return 0, nil, err
+		a.metrics.BackendErrors.Inc()
+		return 0, err
 	}
-	a.metrics.SampleDuration.Set(time.Since(t).Seconds())
 
-	return storageRadius, sample.Hash.Bytes(), nil
+	return time.Duration(timeLimiterBlock.Time) * time.Second / time.Nanosecond, nil
```

Comment:
Not sure about this part `time.Second / time.Nanosecond`. Since `time.Nanosecond` is equal to `1`, this will always return time in seconds. Is that the intention? If so, then `time.Nanosecond` can be omitted.